datafusion/common/src/config.rs

```diff
@@ -252,6 +252,11 @@ config_namespace! {
         /// string length and thus DataFusion can not enforce such limits.
         pub support_varchar_with_length: bool, default = true
 
+        /// If true, permit `VARCHAR` default convert to `Utf8View` in the logical plan.
+        /// If false, `VARCHAR` will be converted to `Utf8` in the logical plan.
+        /// Default is false.
+        pub support_varchar_to_view_types: bool, default = false
```

Comment:
I recommend naming this slightly differently to make it clearer what it is doing

```suggestion
        /// If true, `VARCHAR` is mapped to `Utf8View` during SQL planning.
        /// If false, `VARCHAR` is mappped to `Utf8`  during SQL planning.
        /// Default is false.
        pub default_varchar_views: bool, default = false
```

Reply:
Thank you @alamb for good suggestion, addressed in latest PR.